include/boost/unordered/detail/implementation.hpp

```diff
@@ -2414,7 +1761,7 @@ namespace boost {
           if (size_ > 0) {
             key_equal pred = this->key_eq();
             for (node_pointer* pp = boost::addressof(itb->next); *pp;
-                pp = boost::addressof((*pp)->next)) {
+                 pp = boost::addressof((*pp)->next)) {
```

Comment:
I was trying to minimize the overall diffs but we were forced to use `std::addressof` in one spot because the Core impl is bugged on gcc-5 and gcc-6.

For stylistic consistency, I'll replace all usages of `boost::addressof` with their std counterpart.